src/main/java/org/jenkinsci/plugins/workflow/cps/CpsBodyExecution.java

```diff
@@ -318,57 +310,37 @@ public Next receive(Object o) {
     }
 
     /**
-     * Creates a sub-context to call {@link BodyExecutionCallback}.
-     * If {@link #createBodyBlockNode} is false, then we don't have distinctive
-     * {@link FlowNode}, so we just hand out the master context.
-     */
-    private StepContext subContext(FlowNode n) {
-        if (n==null)
-            return context;
-        else
-            return new CpsBodySubContext(context,n);
-    }
-
-    /**
      * Inserts the flow node that indicates the beginning of the body invocation.
      *
      * @see #addBodyEndFlowNode()
      */
-    private @CheckForNull StepStartNode addBodyStartFlowNode(FlowHead head) {
-        if (createBodyBlockNode) {
-            StepStartNode start = new StepStartNode(head.getExecution(),
-                    context.getStepDescriptor(), head.get());
-            this.startNodeId = start.getId();
-            start.addAction(new BodyInvocationAction());
-            head.setNewHead(start);
-            return start;
-        } else {
-            return null;
-        }
+    private @Nonnull StepStartNode addBodyStartFlowNode(FlowHead head) {
+        StepStartNode start = new StepStartNode(head.getExecution(),
+                context.getStepDescriptor(), head.get());
+        this.startNodeId = start.getId();
+        start.addAction(new BodyInvocationAction());
+        head.setNewHead(start);
+        return start;
     }
 
     /**
      * Inserts the flow node that indicates the beginning of the body invocation.
      *
      * @see #addBodyStartFlowNode(FlowHead)
      */
-    private @CheckForNull StepEndNode addBodyEndFlowNode() {
-        if (createBodyBlockNode) {
-            try {
-                FlowHead head = CpsThread.current().head;
-
-                StepEndNode end = new StepEndNode(head.getExecution(),
-                        getBodyStartNode(), head.get());
-                end.addAction(new BodyInvocationAction());
-                head.setNewHead(end);
-
-                return end;
-            } catch (IOException e) {
-                LOGGER.log(WARNING, "Failed to grow the flow graph", e);
-                throw new Error(e);
-            }
-        } else {
-            return null;
+    private @Nonnull StepEndNode addBodyEndFlowNode() {
+        try {
+            FlowHead head = CpsThread.current().head;
+
+            StepEndNode end = new StepEndNode(head.getExecution(),
```

Comment:
Ask whoever wrote the code originally, @kohsuke I suppose. I was merely deleting an `if`-block.